### PR TITLE
Khalil/1741 baselevel protections metrics

### PR DESCRIPTION
**Pull request by kc1116**
This PR adds 2 metrics:
1. _unauthorized_messages_count_ - number of messages that failed authorization validation 
2. _rate_limited_unicast_messages_count_ - number of messages sent via unicast that have been rate limited

Cleanup authorized sender validator and usage.

### CODE REVIEW

**Review by peterargue — APPROVED** [review]
one comment about the subsystem string, but otherwise looks good

File: module/metrics/namespaces.go [hunk 1]
```diff
@@ -19,11 +19,13 @@ const (
...
-	subsystemQueue   = "queue"
-	subsystemDHT     = "dht"
-	subsystemBitswap = "bitswap"
+	subsystemGossip       = "gossip"
+	subsystemEngine       = "engine"
+	subsystemQueue        = "queue"
+	subsystemDHT          = "dht"
+	subsystemBitswap      = "bitswap"
+	subsystemAuth         = "authorization"
+	subsystemRateLimiting = "rate_limit"
```
Comment (yhassanzadeh13) [inline module/metrics/namespaces.go:28]:
```suggestion
	subsystemRateLimiting = "ratelimit"
```

File: network/slashing/consumer.go [hunk 2]
```diff
@@ -18,27 +21,45 @@ const (
...
+func NewSlashingViolationsConsumer(log zerolog.Logger, metrics module.NetworkSecurityMetrics) *Consumer {
+	return &Consumer{
+		log:     log.With().Str("module", "network_slashing_consumer").Logger(),
+		metrics: metrics,
+	}
 }
 
 func (c *Consumer) logOffense(networkOffense string, violation *Violation) {
+	// if violation fails before the message is decoded the violation.MsgType will be unknown
+	if violation.MsgType == "" {
```
Comment (yhassanzadeh13) [inline network/slashing/consumer.go:38]:
```suggestion
	if len(violation.MsgType) == 0 {
```

File: network/validator/authorized_sender_validator.go [hunk 3]
```diff
@@ -20,63 +20,74 @@ var (
...
+		// don't crash as a result of external inputs since that creates a DoS vector
+		av.log.Error().
+			Err(err).
+			Str("peer_id", from.String()).
+			Str("role", identity.Role.String()).
+			Str("peer_node_id", identity.NodeID.String()).
+			Str("message_type", msgType).
+			Bool("unicast_message", isUnicast).
+			Msg("unexpected error during message validation")
+		return msgType, err
```
Comment (yhassanzadeh13) [inline network/validator/authorized_sender_validator.go:83]: Please also define the metric for `unexpected` error. We may later need to set an alert for this on Grafana.

**Review by yhassanzadeh13 — APPROVED** [review]
LGTM, added a few minor comments.

File: network/slashing/consumer.go (hunk 2 above)
Comment (kc1116) [reply network/slashing/consumer.go]: https://github.com/onflow/flow-go/pull/3146/commits/401014db3f4d24664f9f795428443b3129fa84b0

File: network/validator/authorized_sender_validator.go (hunk 3 above)
Comment (kc1116) [reply network/validator/authorized_sender_validator.go]: https://github.com/onflow/flow-go/pull/3146/commits/70be5b17cdd4fc8ebf79b6199459c5af46da6165

**Comment by kc1116** [comment]
bors merge

**Pull request merged**